pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go

```diff
@@ -117,19 +117,28 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 		}
 	}
 
-	if len(images) == 0 {
+	switch {
+	case len(images) == 0:
 		syncCtx.Recorder().Warningf("MissingVersion", "no image found for operand pod")
-	} else if len(images) > 1 {
+
+	case len(images) > 1:
 		syncCtx.Recorder().Eventf("MultipleVersions", "multiple versions found, probably in transition: %v", strings.Join(images.List(), ","))
-	} else {
-		c.versionRecorder.SetVersion(
-			c.operandName,
-			status.VersionForOperandFromEnv(),
-		)
-		c.versionRecorder.SetVersion(
-			"operator",
-			status.VersionForOperatorFromEnv(),
-		)
+
+	default: // we have one image
+		// if have a consistent image and if that image the same as the current operand image, then we can update the version to reflect our new version
+		if images.List()[0] == status.ImageForOperandFromEnv() {
+			c.versionRecorder.SetVersion(
+				c.operandName,
+				status.VersionForOperandFromEnv(),
+			)
+			c.versionRecorder.SetVersion(
+				"operator",
+				status.VersionForOperatorFromEnv(),
```

Comment:
Is this setting the `operator` version for the operator as a whole, or just for this one static pod controller (with the operator's `operator` version leveling after all its sub-controllers have leveled)?

Reply:
> Is this setting the `operator` version for the operator as a whole, or just for this one static pod controller (with the operator's `operator` version leveling after all its sub-controllers have leveled)?

it's voting. anything needing a different version sets it separately.  But static pod operators don't get choices, it changes when their static pods are up to date.

Reply:
> ...it's voting. anything needing a different version sets it separately...

Is there aggregation logic that tallies votes somewhere, similar to what `UnionClusterCondition` does for conditions?  Grepping for `"operator"` is not turning up an aggregator for me...

Reply:
For what it's worth, I'd expect each controller to set an `operandName`, as you have a bit further up here.  And then operator-level logic that looks at all the controllers and bumps `operator` when they all match their target operand version.